controllers/comment.js

```diff
@@ -1,22 +1,56 @@
 const commentService = require('~/services/comment')
+const Comment = require('~/models/comment')
+const errors = require('~/consts/error').errors
 
 const addComment = async (req, res) => {
-  const { id: author, role: authorRole } = req.user
-  const data = req.body
+  const { id: authorId, role: authorRole } = req.user
+  const { text } = req.body
   const { id: cooperationId } = req.params
 
-  const comment = await commentService.addComment({ text: data.text, author, authorRole, cooperationId })
+  try {
+    const comment = await commentService.addComment({
+      text,
+      author: authorId,
+      authorRole,
+      cooperationId
+    })
+    const populatedComment = await Comment.findById(comment._id).populate({
+      path: 'author',
+      select: 'firstName lastName'
+    })
 
-  res.status(201).json(comment)
+    if (!populatedComment.author) {
+      return res.status(404).json(errors.USER_NOT_FOUND)
+    }
+
+    const response = {
+      _id: populatedComment._id,
+      text: populatedComment.text,
+      author: {
+        _id: populatedComment.author._id,
+        firstName: populatedComment.author.firstName,
+        lastName: populatedComment.author.lastName
+      },
+      cooperation: populatedComment.cooperation,
+      createdAt: populatedComment.createdAt,
+      updatedAt: populatedComment.updatedAt
+    }
+
+    res.status(201).json(response)
+  } catch (error) {
+    res.status(500).json({ ...errors.INTERNAL_SERVER_ERROR, details: error.message })
+  }
 }
 
 const getComments = async (req, res) => {
-  const { id } = req.user
   const { id: cooperationId } = req.params
 
-  const comments = await commentService.getComments(cooperationId, id)
-
-  res.status(200).json(comments)
+  try {
+    const comments = await commentService.getComments(cooperationId, req.user.id)
+    res.status(200).json(comments)
+  } catch (error) {
+    res.status(500).json({ ...errors.INTERNAL_SERVER_ERROR, details: error.message })
+  }

```

Comment:
Is it good idea send only `500` status-codes to client?

Reply:
Sometimes it can be server error (as example if it does not connected to DB).
But sometimes it can be another kinds of errors, as example user do not have rights to get comments in other users co-operations. We can use `403 Forbidden` for case, that described in issue.